pluggy.py

```diff
@@ -659,7 +688,9 @@ def varnames(func):
         return ()
 
     args, defaults = spec.args, spec.defaults
-    args = args[:-len(defaults)] if defaults else args
+    index = -len(defaults) if defaults else None
+    args, defaults = args[slice(0, index)], args[slice(
```

Comment:
the expression is magical enough that i have to think very hard to follow it by now
and this can be a very simple if .. else  thats really easy to follow